### DISCUSSION TITLE
[Feature request] Alternate images for selected items

### DISCUSSION THREAD

**Discussion by zpaolo11x** [complexlogic/flex-launcher#12] [Ideas]
Not sure if this is the right place to put an idea or feature request, btw, would it be possible to add an option where a different icon is defined and shown only when the item is selected? So when the user highlights an item on the screen a different image (brighter, highlited, whatever) is shown...

**Comment by complexlogic** [comment]
It would be relatively easy to implement in the program. My main concern is how it should be added to the config file. If this feature is added, it should be optional only. Currently, the menu entries are parsed as:
```
Entry=title;icon_path;command
```
The program gets 3 tokens delimited by the ; character. It needs to be an exact number amount, so making it 3 _or_ 4 is not possible. You could work around this by further tokenizing the second token with a different character. For example, a comma separating the normal and highlighted icon:
```
Entry=title;normal_icon,selected_icon;command
```
Then, if the comma is absent it would be like normal. So it would be mostly backwards compatible, but it would add a restriction that the icon path itself can't contain a comma, as it would interfere with the parsing.

Another option is to add a suffix such as `_selected` to the filename, and have the program check for it. For example, if the icon path is given as `/path/to/kodi.png`, then the program will check for `/path/to/kodi_selected.png` and, if it exists, it will use that as the selected icon.

The second option would not require any changes to the menu specification, so I am favoring that one somewhat. It is important if this feature is added that it is optional and does not break backwards compatibility.

Any other ideas?

**Reply by zpaolo11x** [reply]
I completely understand the tokenizing issue, I think the _selected suffix is a nice idea, probably the easiest way to implement this, and from a user perspective one already needs a way to distinguish "plain" to "selected" images. The fact that one doesn't need to change the config to add this is another added benefit, I vote for that :)

**Reply by complexlogic** [reply]
Ok. It's a mostly straightforward change. I'll take a look at it and upload an experimental branch sometime in the next couple weeks.

**Comment by complexlogic** [comment]
@zpaolo11x Check out the new branch `selected_icon`:
```
git clone -b selected_icon https://github.com/complexlogic/flex-launcher.git
```
As previously said, you add `_selected` to the end of the icon filename to make it override as the selected icon.

Post some screenshots, I am curious to see what your proposed design looks like.

**Comment by zpaolo11x** [comment]
I just compiled the source and it seems to work perfectly! I'm still thinking of how I want it to look, at the moment a first example is with a white "neon" border around application icons, see attached screenshot :)

Other ideas could be "zooming" icons with a smaller and larger version, or different images for selected and unselected (Amiga style) the possibilities are endless :D 

<img width="567" alt="Screenshot 2022-05-15 at 13 19 07" src="https://user-images.githubusercontent.com/35961910/168470189-c4295690-c2a2-430d-bcb6-475e53e72533.png">

**Reply by complexlogic** [reply]
Wow, that looks very impressive. Great job! I'm glad to hear the feature is working. I will merge it before the next release.

**Comment by Veda-I** [comment]
I like this version, I will test it when I receive my HTPC on monday. One additional feature request related to this one:

option to have a little sound when you navigate from one item to another (could give more multimedia style imo :) )
